implementation/src/main/java/io/smallrye/config/EnvConfigSource.java

```diff
@@ -133,4 +140,16 @@ private static int getEnvOrdinal() {
 
         return DEFAULT_ORDINAL;
     }
+
+    protected Object writeReplace() {
+        return new Ser();
+    }
+
+    static final class Ser implements Serializable {
+        private static final long serialVersionUID = 6812312718645271331L;
+
+        protected Object readResolve() {
+            return new EnvConfigSource();
+        }
```

Comment:
```suggestion
    Object writeReplace() {
        return new Ser();
    }

    static final class Ser implements Serializable {
        private static final long serialVersionUID = 6812312718645271331L;

        Object readResolve() {
            return new EnvConfigSource();
        }
```

Serialization replacement methods should not be visible to subclasses or anyone outside of the API.